src/app/pages/admin/private/shop-details/shop-details.component.ts

```diff
@@ -298,4 +301,32 @@ export class ShopDetailsComponent extends BaseComponent {
   acceptTermsAndConditions() {
     this.termsAndCondition = !this.termsAndCondition;
   }
+
+  getShopDetails() {
+    this.shopService.getShopDetails(this.shopId)
+      .then(response => this.initializeForm(response['data']['shop']))
+      .catch(error => {
+        this.router.navigate([APP_ROUTES.DASHBOARD])
+      })
+  }
+
+  updateShopActiveStatus() {
+    this.isShopActive = !this.isShopActive;
```

Comment:
this is wrong, you shouldn't update local value before hitting API request & fetching actual response